redis/src/main/scala/zio/redis/options/Connection.scala

```diff
@@ -82,11 +148,269 @@ trait Connection {
+          .orDie
+      )
+
+    override def decoder[A](schema: Schema[A]): ZTransducer[Any, String, Byte, A] =
+      ZTransducer.utfDecode >>> ZTransducer.fromFunctionM((s: String) =>
+        ZIO.fromEither(decode(schema)(Chunk.fromArray(s.getBytes)))
+      )
+
+    override def encode[A](schema: Schema[A]): A => Chunk[Byte] = { in =>
+      val values = schema.toDynamic(in)
+      val res    = encodeValue(schema, values)
+
+      Chunk.fromArray(res.getBytes)
+    }
+
+    override def decode[A](schema: Schema[A]): Chunk[Byte] => Either[String, A] =
+      in => {
+        val params = paramsToDynamicValue(schema, new String(in.toArray))
+
+        params.flatMap(dv => schema.fromDynamic(dv))
+      }
+
+    private def encodeValue[A](schema: Schema[A], v: DynamicValue): String =
+      (schema, v) match {
+        case (Schema.Transform(f, _, _, _, _), _) => encodeValue(f, v)
+        case (Schema.GenericRecord(_, fields, _), DynamicValue.Record(_, values)) =>
+          fields.toChunk.map { f =>
+            val value = values(f.label)
+            val title = fieldNames.find(_._2 == f.label).map(_._1).getOrElse(f.label)
+
+            s"$title=${encodeValue(f.schema, value)}"
+          }
+            .mkString(" ")
+        case (Schema.CaseClass1(_, _, _, extractF, _), DynamicValue.Record(_, _)) =>
+          val v1 = schema.fromDynamic(v).toOption.map(extractF)
+
+          v1.map(_.toString).getOrElse("")
+        case (Schema.CaseClass2(_, _, _, _, extractF1, extractF2, _), DynamicValue.Record(_, _)) =>
+          val obj = schema.fromDynamic(v).toOption.get
+
+          val v1 = extractF1(obj)
+          val v2 = extractF2(obj)
+
+          s"$v1$v2"
+        case (Schema.Lazy(s), _)         => encodeValue(s(), v)
+        case (_, DynamicValue.NoneValue) => ""
+        case (
+              Schema.Primitive(StandardType.DurationType, _),
+              DynamicValue.Primitive(d: java.time.Duration, _)
+            ) =>
+          d.getSeconds.toString
+        case (_, DynamicValue.SomeValue(value))          => encodeValue(schema, value)
+        case (_, DynamicValue.Primitive(None, _))        => ""
+        case (_, DynamicValue.Primitive(Some(value), _)) => value.toString
+        case (_, DynamicValue.Primitive(value, _))       => value.toString
+        case x                                           => throw new IllegalArgumentException(s"Unsupported schema: $x $v")
+      }
+
+    private def createDynamicValue(value: DynamicValue, optional: Boolean): DynamicValue =
+      if (optional) DynamicValue.SomeValue(value) else value
+
+    // Creates a dynamicValue from a string and a standardType
+    private def parseDynamicValue[A](
+      value: String,
+      standardType: StandardType[A],
+      optional: Boolean
+    ): Either[String, DynamicValue] = standardType match {
+      case _ if value == "" && optional => Right(DynamicValue.NoneValue)
```

Comment:
Putting it in the back might be a little better?

Reply:
Sorry for that, non native english speaker, what do you mean with this 'Putting it in the back' ? Are you referring to the default case _ ? (sorry again for my clumsiness)

Reply:
> Are you referring to the default case _ ? (sorry again for my clumsiness)
Yes.
(Sorry, I don't speak English very well. 🤕 )

Reply:
It is the first case to be evaluated to exclude compare any type when empty input value is present,
it might be clearer to exclude it with a previous if before the match ...

Reply:
👌🏻